include/h2o.h

```diff
@@ -1809,15 +1819,15 @@ inline void h2o_req_unsetenv(h2o_req_t *req, const char *name, size_t name_len)
     req->env.size -= 2;
 }
 
-inline int h2o_pull(h2o_req_t *req, h2o_ostream_pull_cb cb, h2o_iovec_t *buf)
+inline enum h2o_stream_send_state h2o_pull(h2o_req_t *req, h2o_ostream_pull_cb cb, h2o_iovec_t *buf)
 {
-    int is_final;
+    enum h2o_stream_send_state stream_state;
     assert(req->_generator != NULL);
-    is_final = cb(req->_generator, req, buf);
+    stream_state = cb(req->_generator, req, buf);
     req->bytes_sent += buf->len;
-    if (is_final)
+    if (h2o_stream_send_state_is_final(stream_state))
```

Comment:
How about doing either of the following?
* change the test to `state != H2O_SEND_STATE_IN_PROGRESS`
* introduce is_in_progress function instead of is_final function, and call the function

The reason I suggest this is because having a is_final function that tests if the state is either final or reset is counterintuitive, and also because doing a logical in this hot path is something I'd be happy to avoid.